### ISSUE TITLE
Read in map extremely slow when dealing with large Payload

### ISSUE THREAD

**Issue by jeremyalbrecht** [hazelcast/hazelcast-python-client#434]
Hi, I kind of have a performance issue when using the Python client to retrieve very large objects (16Mbytes protobuf object for example).

There is no problem when setting the value in the cluster it takes a reasonable amount of time, but when it comes to reading it back it takes between 7 and 10 seconds.

I investigated this performance issue, my first idea was that the cluster member had difficulties to give back such a large item. That is why I implemented the same code using the Java (Scala) client: the performance issue does not appear and the reading less time than writing and is really short. In Scala I have the desired behaviour.

I then thought of a performance issue with Python, I implemented in Python using the requests module a way of writing and getting objects in a Map using the REST API. Again, with this solution, it takes more time than using the python-client for writing but there is no performance issue when reading an object.

This table compares the performance of these implementations (each figure is a mean of 100 measurements) :
| Protobuf object size | Python write | Python read | Scala write | Scala read | Python REST API write | Python REST API read |
|----------------------|--------------|-------------|-------------|------------|-----------------------|----------------------|
| 59 bytes             | 1ms          | 0.75ms      | 0.74ms      | 0.44ms     |                       |                      |
| 4 Megabytes          | 86ms         | 8681ms      | 68ms        | 20ms       |                       |                      |
| 16 Megabytes         | 337ms        | 8834ms      | 290ms       | 82ms       | 389ms                 | 155ms                |

Is there any reason why the reading in a Python Map takes more than 8 sec (!!!) to get a 16Mo object ? 

I tried to dig into the source code of the python client without finding somehing relevant. The delay seems to happen within the _send method of InvocationService.

Here is a sample code to reproduce the issue
```python
import time
import hazelcast
import numpy as np
import pandas as pd


MAP_NAME = "source_debug"

def random_picture():
    image = np.random.random((1000, 1000))
    picture = image.tobytes()
    return picture

def put_in_map(map, id, content):
    b = content
    start = time.time()
    map.set(id, bytearray(b))
    stop = time.time()
    print("Storing key({}) done in {}ms".format(len(b),
                                                   (stop - start) * 1000))
    return len(b), (stop -start) * 1000


def get_and_assert(map, id, content):
    start = time.time()
    b = bytes(map.get(id))
    stop = time.time()
    assert b == content
    print("Getting key ({}) done in {}ms".format(len(b),
                                                    (stop - start) * 1000))
    return (stop - start) * 1000


hazelcast = hazelcast.HazelcastClient()
map = hazelcast.get_map(MAP_NAME).blocking()

speed_picture = random_picture()

put_in_map(map, "test1", speed_picture)
get_and_assert(map, "test1", speed_picture)
put_in_map(map, "camera", speed_picture)
get_and_assert(map, "camera", speed_picture)

```

**TL;DR;** Python client takes more than 8 seconds to retrieve a value from a Hazelcast Map, whereas using the REST API in Python, or using the Java client, the same operation takes around 100ms

**Comment by mdumandag** [comment]
Hi @jeremyalbrecht. Thanks a lot for the detailed issue report. I am able to reproduce the problem as well. 

I will look into this today and get back to you

**Comment by mdumandag** [comment]
Hi again @jeremyalbrecht.

I have identified the issue and prepared a fix. We should probably merge it today. Since this is a critical bug, we will release a patch version, probably this week, after completing our regular release procedure, which takes a couple of days.

Until then, you can use the code in the master branch after merging #436.

**Comment by jeremyalbrecht** [comment]
Hi @mdumandag,
Thanks a lot for your reactivity in this issue. I tested your changes and it works like a charm ! 
I will directly use the code in the PR.

Have a nice day

**Comment by mdumandag** [comment]
Great to hear that. Thanks again for raising this issue.

Have a nice day
